syntax = "proto3";

package Ydb.Config.V1;

option java_package = "com.yandex.ydb.bsconfig.v1";
option java_outer_classname = "ConfigGrpc";
option java_multiple_files = true;

import "ydb/public/api/protos/ydb_config.proto";

service ConfigService {

    // Initialize Blobstorage/single config
    rpc ReplaceConfig(Config.ReplaceConfigRequest) returns (Config.ReplaceConfigResponse);

    // Fetch Blobstorage/single config
    rpc FetchConfig(Config.FetchConfigRequest) returns (Config.FetchConfigResponse);

    // Bootstrap automatically configured cluster
    rpc BootstrapCluster(Config.BootstrapClusterRequest) returns (Config.BootstrapClusterResponse);

}
